LiskHQ/lisk-desktop
# Show correct Hardware wallet errors for transactions

### What was the problem?

This PR resolves #4882
Review [this PR](https://github.com/LiskHQ/lisk-desktop/pull/4985) before this one.

### How was it solved?

- Adding 3 error state (disconnected, rejected, Lisk app closed) for hardware wallet when initialing a transaction.
- Added UX to prevent user from initialing a transaction when we know it wont work.
- Removed dead code

### How was it tested?
Manually these 3 error states:
1. Connect HW device, enter pin code, dont open the Lisk app. Try to make a transaction. 
   - Expected: Send button to be disabled and show error label with "Open the Lisk app to continue". Open Lisk app on device, send button should now be enabled. 
2. Enter send flow when "Confirm your transaction" modal pops up, disconnect the device.
   - Expected: A modal showing "Device disconnected"
3. Enter send flow when "Confirm your transaction" modal pops up, reject the transaction on the device.
   - Expected: A modal showing "Transaction aborted on device"
